### PR TITLE
Fix not to set chunked encoding header when 304 response is sent

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
When an HTTP/1.1 304(Not Modified) response is sent, `Transfer-Encoding: chunked` header cannot be sent
because the content of the response is always empty.
https://datatracker.ietf.org/doc/html/rfc7232#section-4.1
```
A 304 response cannot contain a message-body; it is always terminated
by the first empty line after the header fields.
```

Modification:
- Do not set chunked encoding header when the content of the response is always empty.

Result:
- Armeria server no longer sends chunked encoding header with 304(Not Modified) status.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java [hunk 1]
```diff
@@ -126,7 +126,9 @@ private HttpResponse convertHeaders(ResponseHeaders headers, boolean endStream,
         } else {
             res = new DefaultHttpResponse(HttpVersion.HTTP_1_1, nettyStatus, false);
             convertHeaders(headers, res.headers(), isTrailersEmpty);
-            maybeSetTransferEncoding(res);
+            if (!HttpStatus.isContentAlwaysEmpty(statusCode)) {
+                maybeSetTransferEncoding(res);
+            }
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java:131]: Should we instead modify `maybeSetTransferEncoding()` to require a status code as an additional parameter?

File: core/src/test/java/com/linecorp/armeria/client/Http1HeaderNamingTest.java [hunk 2]
```diff
@@ -139,7 +139,6 @@ void serverTraditionalHeaderNaming(boolean useHeaderNaming) throws IOException {
             boolean hasXForwardedFor = false;
             for (;;) {
                 final String line = reader.readLine();
-                System.out.println(line);
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/client/Http1HeaderNamingTest.java:142]: Thanks!

**Review by trustin — COMMENTED** [review]
Thanks a lot for the fix, @minwoox ! 🙇

File: core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java (hunk 1 above)
Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java]: Thanks! updated. 😉

**Comment by minwoox** [comment]
Thanks for reviewing guys. 😉

**Pull request merged**